capstone/client/src/components/ListPage.js

```diff
@@ -115,7 +122,28 @@ class ListPage extends Component {
    * @TODO Change this function to make a GET request to obtain store recommendations based 
    * on the selected items.
    */
-  onSubmit = () => {
+   onSubmit = async () => {
+    var latit = this.state.location.latitude;
+    var longi = this.state.location.longitude;
+    if(this.state.zipCodeRequired && (!this.state.zipCode || this.state.zipCode.length !== 5 || isNaN(this.state.zipCode))){
+      this.setState({
+        errorMessage: "Invalid Zip Code"
+      });
+      return;
+    }
+    try {
+      let response = await Geocode.fromAddress(this.state.zipCode);
+      const { lat, lng } = response.results[0].geometry.location;
+      latit = lat;
```

Comment:
Why do you need to keep these two variables: latit and longi? Can't you just call this.state.location.latitude where you need it?

Reply:
Because this.state.location.latitude does not update immediately after you set it in react, it takes some amount of cycles so you have to use a separate variable